compiler/compiler.go

```diff
@@ -383,6 +391,12 @@ func apply(sub map[uint64]semantic.MonoType, props []semantic.PropertyType, t se
 			return t
 		}
 		return semantic.NewFunctionType(apply(sub, nil, retn), args)
+	case semantic.Vec:
+		element, err := t.ElemType()
+		if err != nil {
+			return t
+		}
+		return semantic.NewVectorType(apply(sub, nil, element))
```

Comment:
I'm a little unclear on what the `props` are in this function. I can see that the `Arr` case propagates them, so maybe `Vec` should too? I'd like to understand what this parameter is for. What it's doing doesn't seem quite right to me.